python-manual/modules/ROOT/pages/data-types.adoc

```diff
@@ -0,0 +1,411 @@
+= Data types and mapping to Cypher types
+
+The tables in this section show the mapping between Cyper data types and Python types.
+
+== Core data types
+
+[options="header"]
+|===
+| Cypher Type | Python Type
+| <<null>>    | `None`
+| List        | `list`
+| Map         | `dict`
+| Boolean     | `bool`
+| Integer     | `int`
+| Float       | `float`
+| String      | `str`
+| ByteArray   | `bytearray`
+|===
+
+For full documentation, see link:{neo4j-docs-base-uri}/api/python-driver/current/api.html#core-data-types[API documentation -- Core data types].
+
+
+== Temporal types
+
+Temporal data types are implemented by the `neo4j.time` module. It provides a set of types compliant with ISO-8601 and Cypher, which are similar to those found in Python's native `datetime` module. To convert between driver and native types, use the methods `.from_native()` and `.to_native()` (does not apply to `Duration`).
+
+Sub-second values are measured to nanosecond precision and the types are mostly compatible with link:https://pypi.org/project/pytz/[pytz]. Some timezones (e.g., `pytz.utc`) work exclusively with the built-in `datetime.datetime`.
+
+[options="header"]
+|===
+| Cypher Type   | Python Type
+| Date          | `neo4j.time.Date`
+| Time          | `neo4j.time.Time`†
+| LocalTime     | `neo4j.time.Time`††
+| DateTime      | `neo4j.time.DateTime`†
+| LocalDateTime | `neo4j.time.DateTime`††
+| Duration      | `neo4j.time.Duration`
+|===
+
+† Where `tzinfo` is not `None`. +
+†† Where `tzinfo` is `None`.
+
+.How to use temporal types in queries
+[source,python]
+----
+from neo4j import GraphDatabase
+from datetime import datetime
+import pytz
+from neo4j.time import DateTime
+
+
+URI = "neo4j://localhost:7687"
+AUTH = ("neo4j", "secretgraph")
+
+
+friends_since = DateTime(year=1999, month=11, day=23,
+                         hour=7, minute=47, nanosecond=4123)
+friends_since = pytz.timezone("US/Eastern").localize(friends_since)
+
+# Python's native datetimes work as well.
+# They don't support the full feature-set of Neo4j's type though.
+# py_friends_since = datetime(year=1999, month=11, day=23, hour=7, minute=47)
```

Comment:
Since it wouldn't match exactly anyway (because of the 123 nanoseconds difference), I think we can skip the extra parameter. Reopen if you don't feel so.